Add linting.yml

neon60 opened the pull request:
- The following PR has to merged in first: https://github.com/ROCm/rocm-docs-core/pull/794
- Added spellcheck to docs folder md and rst files
- Added markdown file check
- Fixed the makrdown styling inconsistency
- Fixed abbreviation usage inconsistency
- Fixed typos

yhuiYH reviewed (approved):
Nice, great effort here to fix all the typos.  I like this.  Let's merge this thru gerrit, and back out to GitHub.

neon60 committed:
Add linting and spellcheck

- Markdown fixes
- Fix spellcheck
- Clean up
- Supress etoc warning
- Remove folder before clone if it's exist
- Change liniting.yml to use develop

neon60 commented:
> Nice, great effort here to fix all the typos. I like this. Let's merge this thru gerrit, and back out to GitHub.

Thanks. Squashed the commits, merging to docs/develop. It will reach amd-staging soon.
